core/src/test/java/org/elasticsearch/index/query/AbstractQueryTestCase.java

```diff
@@ -318,6 +319,25 @@ public void testFromXContent() throws IOException {
     }
 
     /**
+     * Generic test that first adds a "boost" field to the otherwise randomly created query and
+     * then replaces the field with an unknown field. This should trigger ParsingException that
+     * points out the unknown field.
+     * Queries that allow arbitrary field names at this level need to override this test so it
+     * doesn't get executed for those specific tests.
+     */
+    public void testUnknownField() throws IOException {
+        QB testQuery = createTestQueryBuilder();
+        testQuery.boost(1.0f);
+        String queryAsString = testQuery.toString().replace("boost", "bogusField");
```

Comment:
Thanks, this unfortunately is true and shows that the test approach is maybe still a bit too brittle. I just realized that maybe there are boost fields in subqueries which are replaced here instead of the one on the "main" level in the test query. This seemed to work reasonably well because its a field that (almost) all queries should have on the same level as the other more specific fields. Hard to think about a generic way of breaking things in a predictable way, but any suggestions welcome.